ffi/passmanagers.cpp

```diff
@@ -7,6 +15,49 @@
 
 using namespace llvm;
 
+
+/* Helpers to override the diagnostic handler when running
+ * optimizations.
+ * The default diagnostic handler prints some remarks unconditionally,
+ * see http://lists.llvm.org/pipermail/llvm-dev/2016-July/102252.html
+ */
+typedef std::tuple<LLVMContext::DiagnosticHandlerTy, void *> diag_handler_t;
+
+static diag_handler_t
+SetOptimizationDiagnosticHandler(LLVMContext &Ctx)
+{
+    auto diagnose = [] (const DiagnosticInfo &DI, void *c) {
+        // If an error, print the message and bail out (as the default
+        // handler does).
+        DiagnosticSeverity DS = DI.getSeverity();
+        if (DS == DS_Error) {
+            raw_ostream &out = errs();
+            DiagnosticPrinterRawOStream DP(out);
+            out << "LLVM error: ";
+            DI.print(DP);
+            out << "\n";
+            exit(1);
```

Comment:
Can we signal the python side instead of killing the process here?  `LLVMPY_RunFunctionPassManager` returns something to indicate success/failure already.  I am just afraid that calling `exit()` here makes it harder to debug if things go wrong.